pages/History.qml

```diff
@@ -34,342 +34,1519 @@ import moneroComponents.WalletManager 1.0
-                        color: "#404040"
+                    MouseArea {
+                        enabled: parent.enabled
+                        anchors.fill: parent
+                        hoverEnabled: true
+                        cursorShape: Qt.PointingHandCursor
+                        onClicked: {
+                            root.paginationPrevClicked();
+                        }
                     }
+                }
+
+                Rectangle {
+                    id: paginationNext
+                    Layout.preferredWidth: 18 * scaleRatio
+                    Layout.preferredHeight: 20 * scaleRatio
+                    color: "transparent"
+                    opacity: enabled ? 1.0 : 0.6
+                    enabled: false
 
                     Image {
+                        anchors.verticalCenter: parent.verticalCenter
+                        anchors.right: parent.right
+                        height: 8 * scaleRatio
+                        width: 12 * scaleRatio
+                        source: "../images/whiteDropIndicator.png"
+                        rotation: 270
+                    }
+
+                    MouseArea {
+                        anchors.fill: parent
+                        hoverEnabled: true
+                        cursorShape: Qt.PointingHandCursor
+                        onClicked: {
+                            root.paginationNextClicked();
+                        }
+                    }
+                }
+            }
+        }
+
+        ListView {
+            visible: true
+            id: txListview
+            Layout.fillWidth: true
+            Layout.preferredHeight: contentHeight;
+            model: txListViewModel
+            interactive: false
+
+            delegate: Rectangle {
+                id: delegate
+                property bool collapsed: root.txDataCollapsed.indexOf(hash) >= 0 ? true : false
+                anchors.left: parent.left
+                anchors.right: parent.right
+                height: {
+                    if(!collapsed) return 60;
+                    if(isout && delegate.address !== "") return 320;
+                    return 220;
+                }
+                color: collapsed ? "#06FFFFFF" : "transparent"
+
+                Rectangle {
+                    anchors.top: parent.top
+                    anchors.bottom: parent.bottom
+                    anchors.left: parent.left
+                    width: sideMargin
+                    color: "transparent"
+
+                    Rectangle {
                         anchors.top: parent.top
-                        anchors.left: parent.left
+                        anchors.topMargin: 24 * scaleRatio
+                        anchors.horizontalCenter: parent.horizontalCenter
+                        width: 10 * scaleRatio
+                        height: 10 * scaleRatio
+                        radius: 8 * scaleRatio
+                        color: isout ? "#d85a00" : "#2eb358"
```

Comment:
https://github.com/monero-project/monero-gui/blob/d4d8ff54e6c94f82de7eb012d7b310addeedcbb9/components/HistoryTable.qml#L143

Can we highlight mining reward transactions like before, with a special color?

Reply:
Sorry for not migrating this cool feature. Seems I forgot. Here is how it looks like with mining:

![](https://i.imgur.com/GyFg071.png)